src/components/keypad/index.js

```diff
@@ -9,28 +9,46 @@ import Clickable from "@khanacademy/wonder-blocks-clickable";
 import {Tabbar} from "../tabbar/tabbar";
 import NumericInputPage from "./numeric-input-page";
 import PreAlgebraPage from "./pre-algebra-page";
+import TrigonometryPage from "./trigonometry-page";
 
 import type {Node} from "React";
 import type {TabbarItemType} from "../tabbar/item";
 
-type Props = {
+type Props = {|
     onClickKey: (keyConfig: string) => void,
-};
+    preAlgebra: boolean,
+    trigonometry: boolean,
+|};
 type State = {
     selectedPage: TabbarItemType,
 };
 
+function allPages(props: Props): Array<TabbarItemType> {
+    var pages: Array<TabbarItemType> = ["Numbers"];
+
+    if (props.preAlgebra) {
```

Comment:
Since the only props you're using in here are `preAlgebra` and `trigonometry`, you might want to just pass those two in instead of the whole `props` object. Doesn't make much of a difference though I guess?

Reply:
I think this will evolve to be easier to understand- the props are definitely in an evolving state.